crates/oxc_ast/src/ast/js.rs

```diff
@@ -260,6 +264,8 @@ pub struct IdentifierReference {
     #[cfg_attr(feature = "serde", serde(flatten))]
     pub span: Span,
     pub name: Atom,
+    #[cfg_attr(feature = "serde", serde(skip))]
+    pub reference_id: Option<ReferenceId>,
```

Comment:
@Boshen I don't use `Cell` here, which means we might need to change the api of `semantic` to something like this.

```rust
let semantic_builder = todo!();
let program = todo!();
semantic_builder.resolve_symbol_and_refs(&mut program);
let ret = semantic_builder.build(&program);
```

What's your opinions ?

Reply:
Let's not change APIs in the same PR. I kind of prefer inner immutability here, since it gets the job done without change any other part of the system.